substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -902,7 +925,9 @@ protected abstract class NativeImageDebugBaseMethodInfo extends NativeImageDebug
             // method references in node source positions. So, we do the translation
             // here just to make sure we use a HostedMethod wherever possible.
             method = promoteAnalysisToHosted(m);
-            this.paramInfo = createParamInfo(method);
+            LineNumberTable lineNumberTable = method.getLineNumberTable();
+            line = (lineNumberTable != null ? lineNumberTable.getLineNumber(0) : 0);
```

Comment:
Because when written out in the DWARF info 0 is an illegal line number and -1 is interpreted as the unsigned value 0xffffffff -- n.b. line numbers have to be strictly positive to be valid.